strawberry_django/auth/queries.py

```diff
@@ -1,10 +1,28 @@
 import strawberry_django
 
 
+def get_current_user(info):
+    """Get and return the current user based on various scenarios."""
+    try:
+        user = info.context.request.user
```

Comment:
This can also cause errors when calling this inside an async function

Maybe also provide an `async def aget_current_user(info): return sync_to_async(get_current_user)(info)` bellow for async cases